Actions/PSScriptAnalyzer/action.yml

```
@@ -0,0 +1,62 @@
+name: Run PSScriptAnalyzer
+author: Microsoft Corporation
+branding:
+  icon: "check"
+  color: "gray-dark"
+inputs:
+  path:
+    description: 'Specifies the path to the scripts or module to be analyzed. Wildcard characters are supported.'
+    required: true
+    default: '.\'
+  excludeRule:
+    description: 'Omits the specified rules from the Script Analyzer test. Wildcard characters are supported.'
+    required: false
+  recurse:
+    description: 'Runs Script Analyzer on the files in the Path directory and all subdirectories recursively.'
+    required: false
+  output:
+    description: 'Specifies where the path for the sarif file'
+    required: true
+    default: 'results.sarif'
+runs:
+  using: "composite"
+  steps:
+    - name: Run PSScriptAnalyzer and ConvertToSARIF
+      shell: pwsh
+      run: |
+        $analyzerModule = Get-Module -ListAvailable -Name PSScriptAnalyzer
+        if ($null -eq $analyzerModule) {
+          Install-Module -Name PSScriptAnalyzer -Force
+        }
+
+        $sarifModule = Get-Module -ListAvailable -Name ConvertToSARIF
+        if ($null -eq $sarifModule) {
+          Install-Module -Name ConvertToSARIF -Force
+        }
+        Import-Module -Name ConvertToSARIF -Force
+
+        $htPSA = [ordered]@{ Path = '${{ inputs.path }}'; }
+        Write-Output "Modules installed, now running tests."
+        if(![string]::IsNullOrEmpty('${{ inputs.excludeRule }}')) { $htPSA.add('ExcludeRule', @(${{ inputs.excludeRule }})) }
+        if(![string]::IsNullOrEmpty('${{ inputs.recurse }}')) { $htPSA.add('Recurse', $true) }
+        $htCTS = [ordered]@{ FilePath = '${{ inputs.output }}'; }
+
+        $maxRetries = 3
+        $retryCount = 0
+        $success = $false
+
+        while (-not $success -and $retryCount -lt $maxRetries) {
+          Try {
+            Invoke-ScriptAnalyzer @htPSA -Verbose | ConvertTo-SARIF @htCTS
```

Took a look at one of the runs: https://github.com/microsoft/AL-Go/actions/runs/12711830687/job/35436027525?pr=1388

This seems quite verbose. IMO we can live without that level of verbosity but I'm also okay leaving it in. 

```suggestion
            Invoke-ScriptAnalyzer @htPSA | ConvertTo-SARIF @htCTS
```

---

I agree that it's quite noisy, but I don't think we need it, at least temporary, to debug the random null reference failures.

We can collect some logs on next failure and open an issue on the PSScriptAnalyzer repo, or at least confirm if it's the same issue as here: https://github.com/PowerShell/PSScriptAnalyzer/issues/1867#issuecomment-1340941069